WooCommerce/Classes/ViewRelated/Products/ProductLoaderViewController.swift

```diff
@@ -162,23 +176,29 @@ private extension ProductLoaderViewController {
     }
 
     func presentProductDetails(for product: Product, isEditProductsEnabled: Bool) {
-        let viewController: UIViewController
         if product.productType == .simple && isEditProductsEnabled {
-            viewController = ProductFormViewController(product: product, currency: currency)
+            let productForm = ProductFormViewController(product: product, currency: currency) { viewController in
+                viewController.dismiss(animated: true, completion: nil)
```

Comment:
It seems that this callback is never called.

Reply:
the `onDismiss` closure was called when the user taps on the "discard changes" action 🙂 (it's not called when the user navigates/swipes back)